### PR TITLE
Added ui for fission environments

### PR DESCRIPTION
**Pull request by girishg4t**
Create UI for fission environments, below are the changes
1) Created ui using react and material ui
2) No server side rendering is required
3) Added envconfig.json file for go, dotnet, python, nodejs
4) Added shell script which combines the data from above json into one
5) Used webpack to create production build

As and when the envconfig.json file is added for other environments, no need to make any changes in the ui

### CODE REVIEW

File: environments-ui/src/Environments.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                    <Button size="small" color="primary">
+                      Learn More
+        </Button>
+                  </CardActions> */}
+                                    </Card>
+                                </Grid>
+                            )
+                        }
+                    })
+                }
```
Comment (vishal-biyani) [inline environments-ui/src/Environments.js:171]: If this is not needed, can you please remove it?

File: environments-ui/src/Environments.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                variant="permanent"
+                classes={{
+                    paper: classes.drawerPaper,
+                }}
+            >
+                <Toolbar />
+                <div className={classes.drawerContainer}>
+                    <List dense className={classes.checkBox}>
+                        <ListItem key='all' button>
+                            <ListItemText id='all' primary={`Select all environment`} />
```
Comment (vishal-biyani) [inline environments-ui/src/Environments.js:94]:
```suggestion
                            <ListItemText id='all' primary={`All environments`} />
```

File: environments-ui/src/Environments.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+                                    inputProps={{ 'aria-labelledby': `checkbox-list-secondary-label-all` }}
+                                />
+                            </ListItemSecondaryAction>
+                        </ListItem>
+                        {environments.map((env) => {
+                            const value = env[0].name.split(' ')[0]
+                            const labelId = `checkbox-list-secondary-label-${value}`;
+                            return (
+                                <ListItem key={value} button>
+                                    <ListItemText id={labelId} primary={`Select ${value} environment`} />
```
Comment (vishal-biyani) [inline environments-ui/src/Environments.js:109]:
```suggestion
                                    <ListItemText id={labelId} primary={`${value}`} />
```

File: go/envconfig.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
+[{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.11.4", 
+    "kind": "environment", 
+    "status": "alpha", 
+    "name": "Go Environment",
+    "shortDescription": "It's a Docker image containing a Go runtime, along with a dynamic loader.",
```
Comment (vishal-biyani) [inline go/envconfig.json:7]:
```suggestion
    "shortDescription": "Fission Go environment, which uses dynamic loader based on Go plugins.",
```

File: go/envconfig.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
+[{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.11.4", 
+    "kind": "environment", 
+    "status": "alpha",
```
Comment (vishal-biyani) [inline go/envconfig.json:5]:
```suggestion
    "status": "Stable", 
```

File: go/envconfig.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.11.4", 
+    "kind": "environment", 
+    "status": "alpha", 
+    "name": "Go Environment",
+    "shortDescription": "It's a Docker image containing a Go runtime, along with a dynamic loader.",
+    "readme" : "https://github.com/fission/fission/tree/master/environments/go",
+    "examples" : "https://github.com/fission/fission/tree/master/environments/go",
+    "keywords": [
+        "logging"
```
Comment (vishal-biyani) [inline go/envconfig.json:11]: Not relevant

File: go/envconfig.json [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    "maintainers": [
+        {
+            "name": "nestorsalceda",
+            "link": "https://github.com/nestorsalceda"
+        },
+        {
+            "name": "fedebarcelona",
+            "link": "https://github.com/tembleking"
+        }
+    ]
```
Comment (vishal-biyani) [inline go/envconfig.json:23]: Same as previous comment

File: go/envconfig.json [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+},
+{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.13", 
+    "kind": "environment", 
+    "status": "alpha", 
+    "name": "Go Environment",
+    "shortDescription": "It's a Docker image containing a Go runtime, along with a dynamic loader.",
```
Comment (vishal-biyani) [inline go/envconfig.json:31]:
```suggestion
    "shortDescription": "Fission Go environment, which uses dynamic loader based on Go plugins.",
```

File: go/envconfig.json [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+            "name": "fedebarcelona",
+            "link": "https://github.com/tembleking"
+        }
+    ]
+},
+{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.13", 
+    "kind": "environment", 
+    "status": "alpha",
```
Comment (vishal-biyani) [inline go/envconfig.json:29]:
```suggestion
    "status": "Stable", 
```

File: go/envconfig.json [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+},
+{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.13", 
+    "kind": "environment", 
+    "status": "alpha", 
+    "name": "Go Environment",
+    "shortDescription": "It's a Docker image containing a Go runtime, along with a dynamic loader.",
+    "readme" : "https://github.com/fission/fission/tree/master/environments/go",
+    "examples" : "https://github.com/fission/fission/tree/master/environments/go",
```
Comment (vishal-biyani) [inline go/envconfig.json:33]: Should point to Environments repo

File: go/envconfig.json [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    "keywords": [
+        "logging"
```
Comment (vishal-biyani) [inline go/envconfig.json:35]: Not relevant

File: go/envconfig.json [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+    ],
+    "icon": "https://golang.org/lib/godoc/images/go-logo-blue.svg",
+    "maintainers": [
+        {
+            "name": "nestorsalceda",
+            "link": "https://github.com/nestorsalceda"
+        },
+        {
+            "name": "fedebarcelona",
+            "link": "https://github.com/tembleking"
```
Comment (vishal-biyani) [inline go/envconfig.json:45]: Fission Authors

File: nodejs/envconfig.json [hunk 13]
```diff
@@ -0,0 +1,24 @@
+[{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.0.0",
```
Comment (vishal-biyani) [inline nodejs/envconfig.json:3]: Runtime version is definitely 1.0.0 - Can you please check?

File: nodejs/envconfig.json [hunk 14]
```diff
@@ -0,0 +1,24 @@
+[{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.0.0", 
+    "kind": "environment", 
+    "status": "alpha", 
+    "name": "Nodejs Environment",
+    "shortDescription": "It's a Docker image containing a NodeJS runtime, along with a dynamic loader.  A few common dependencies are included in the package.json file.",
```
Comment (vishal-biyani) [inline nodejs/envconfig.json:7]:
```suggestion
    "shortDescription": "Fission NodeJS environment based on Express with some basic dependencies added",  
```

File: nodejs/envconfig.json [hunk 15]
```diff
@@ -0,0 +1,24 @@
+[{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.0.0", 
+    "kind": "environment", 
+    "status": "alpha",
```
Comment (vishal-biyani) [inline nodejs/envconfig.json:5]:
```suggestion
    "status": "Stable", 
```

File: nodejs/envconfig.json [hunk 16]
```diff
@@ -0,0 +1,24 @@
+[{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.0.0", 
+    "kind": "environment", 
+    "status": "alpha", 
+    "name": "Nodejs Environment",
+    "shortDescription": "It's a Docker image containing a NodeJS runtime, along with a dynamic loader.  A few common dependencies are included in the package.json file.",  
+    "readme" : "https://github.com/fission/fission/blob/master/environments/nodejs",
+    "examples" : "https://github.com/fission/fission/blob/master/environments/nodejs",
```
Comment (vishal-biyani) [inline nodejs/envconfig.json:9]: Should point to Environments repo

File: nodejs/envconfig.json [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+    "keywords": [
+        "logging"
```
Comment (vishal-biyani) [inline nodejs/envconfig.json:11]: Not relevant

File: nodejs/envconfig.json [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+    "icon": "https://nodejs.org/static/images/logos/nodejs-new-pantone-black.svg",
+    "maintainers": [
+        {
+            "name": "nestorsalceda",
+            "link": "https://github.com/nestorsalceda"
+        },
+        {
+            "name": "fedebarcelona",
+            "link": "https://github.com/tembleking"
+        }
```
Comment (vishal-biyani) [inline nodejs/envconfig.json:22]: Please fix

File: python/envconfig.json [hunk 19]
```diff
@@ -0,0 +1,24 @@
+[{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.0.0",
```
Comment (vishal-biyani) [inline python/envconfig.json:3]: We have Python 2 & 3 - so this is incorrect

File: python/envconfig.json [hunk 20]
```diff
@@ -0,0 +1,24 @@
+[{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.0.0", 
+    "kind": "environment", 
+    "status": "alpha", 
+    "name": "Python Environment",
+    "shortDescription": "It's a Docker image containing a Python 3.5 runtime, along with a dynamic loader. A few common dependencies are included in the requirements.txt file.",
```
Comment (vishal-biyani) [inline python/envconfig.json:7]:
```suggestion
    "shortDescription": "Fission Python environment based on Flask framework.",
```

File: python/envconfig.json [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+    ],
+    "icon": "https://upload.wikimedia.org/wikipedia/commons/c/c3/Python-logo-notext.svg",
+    "maintainers": [
+        {
+            "name": "nestorsalceda",
+            "link": "https://github.com/nestorsalceda"
+        },
+        {
+            "name": "fedebarcelona",
+            "link": "https://github.com/tembleking"
```
Comment (vishal-biyani) [inline python/envconfig.json:21]: Similar to previous comments

**Review by vishal-biyani — CHANGES_REQUESTED** [review]
Hey @girishg4t Left a bunch of comments. @sahil-lakhwani Can you please align with this one and give this another review once Girish has fixed things.

@girishg4t  Can we filter by the status too (Alpha/Beta) etc.?

Also, I see something missing such as Python has two env versions - 2 & 3 based, similar thing for NodeJs, Go, JVM etc. so we need to make those entries too.

File: dotnet/envconfig.json [hunk 22]
```diff
@@ -0,0 +1,24 @@
+[{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.0.0", 
+    "kind": "environment", 
+    "status": "alpha", 
+    "name": "Dotnet Environment",
+    "shortDescription": "It's a Docker image containing the dotnet 2.0.0 runtime. The image uses Kestrel with Nancy to host the internal web server and uses Roslyn to compile the uploaded code.",
```
[the remark this answers is not in the document]
Comment (girishg4t) [reply dotnet/envconfig.json]: Changed

File: dotnet/envconfig.json [hunk 23]
```diff
@@ -0,0 +1,24 @@
+[{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.0.0", 
+    "kind": "environment", 
+    "status": "alpha", 
+    "name": "Dotnet Environment",
+    "shortDescription": "It's a Docker image containing the dotnet 2.0.0 runtime. The image uses Kestrel with Nancy to host the internal web server and uses Roslyn to compile the uploaded code.",   
+    "readme" : "https://github.com/fission/fission/tree/master/environments/dotnet20",
+    "examples" : "https://github.com/fission/fission/tree/master/environments/dotnet20",
```
[the remark this answers is not in the document]
Comment (girishg4t) [reply dotnet/envconfig.json]: Changed

File: dotnet/envconfig.json [hunk 24]
```diff
@@ -0,0 +1,24 @@
...
+    "keywords": [
+        "logging"
```
[the remark this answers is not in the document]
Comment (girishg4t) [reply dotnet/envconfig.json]: Changed

File: dotnet/envconfig.json [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+    ],
+    "icon":  "https://upload.wikimedia.org/wikipedia/commons/e/ee/.NET_Core_Logo.svg",
```
[the remark this answers is not in the document]
Comment (girishg4t) [reply dotnet/envconfig.json]: Changed

File: dotnet/envconfig.json [hunk 26]
```diff
@@ -0,0 +1,24 @@
...
+    "maintainers": [
+        {
+            "name": "nestorsalceda",
+            "link": "https://github.com/nestorsalceda"
+        },
+        {
+            "name": "fedebarcelona",
+            "link": "https://github.com/tembleking"
+        }
```
[the remark this answers is not in the document]
Comment (girishg4t) [reply dotnet/envconfig.json]: Added top 3 author from fission repo

File: environments-ui/package.json [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+    "start": "webpack-dev-server --open --mode development",
+    "build": "webpack --mode production"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/girishg4t/fission-ui.git"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
```
[the remark this answers is not in the document]
Comment (girishg4t) [reply environments-ui/package.json]: license: i think would be Apache License 2.0, fission is licensed with apache
author: is who developed it, for now i kept it blank
kerywords: might be react , material ui, fission environments ui etc

File: environments-ui/package.json [hunk 28]
```diff
@@ -0,0 +1,51 @@
...
+  "bugs": {
+    "url": "https://github.com/girishg4t/fission-ui/issues"
+  },
+  "homepage": "https://github.com/girishg4t/fission-ui#readme",
```
[the remark this answers is not in the document]
Comment (girishg4t) [reply environments-ui/package.json]: corrected

File: environments-ui/src/Environments.js [hunk 29]
```diff
@@ -0,0 +1,176 @@
...
+
+import environments from "./resources/environments.json";
+
+const drawerWidth = 240;
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        display: 'flex',
+        height: '300px',
+        width: '200px'
```
[the remark this answers is not in the document]
Comment (girishg4t) [reply environments-ui/src/Environments.js:27]: This are for cards height and width, on mobile it is wrapping correctly only the filter panel remain as it is, need to see what can be done

File: environments-ui/src/Environments.js (hunk 1 above)
Comment (girishg4t) [reply environments-ui/src/Environments.js]: removed

File: environments-ui/src/Environments.js (hunk 2 above)
Comment (girishg4t) [reply environments-ui/src/Environments.js]: Changed

File: environments-ui/src/Environments.js (hunk 3 above)
Comment (girishg4t) [reply environments-ui/src/Environments.js]: Changed

File: go/envconfig.json (hunk 4 above)
Comment (girishg4t) [reply go/envconfig.json]: Changed

File: go/envconfig.json (hunk 5 above)
Comment (girishg4t) [reply go/envconfig.json]: Changed

File: go/envconfig.json (hunk 6 above)
Comment (girishg4t) [reply go/envconfig.json]: Removed

File: go/envconfig.json (hunk 7 above)
Comment (girishg4t) [reply go/envconfig.json]: Done

File: go/envconfig.json (hunk 8 above)
Comment (girishg4t) [reply go/envconfig.json]: Changed

File: go/envconfig.json (hunk 9 above)
Comment (girishg4t) [reply go/envconfig.json]: Changed

File: go/envconfig.json (hunk 10 above)
Comment (girishg4t) [reply go/envconfig.json]: changed

File: go/envconfig.json (hunk 11 above)
Comment (girishg4t) [reply go/envconfig.json]: removed

File: go/envconfig.json (hunk 12 above)
Comment (girishg4t) [reply go/envconfig.json]: changed

File: nodejs/envconfig.json (hunk 13 above)
Comment (girishg4t) [reply nodejs/envconfig.json]: Changed

File: nodejs/envconfig.json (hunk 14 above)
Comment (girishg4t) [reply nodejs/envconfig.json]: Changed

File: nodejs/envconfig.json (hunk 15 above)
Comment (girishg4t) [reply nodejs/envconfig.json]: Changed

File: nodejs/envconfig.json (hunk 16 above)
Comment (girishg4t) [reply nodejs/envconfig.json]: Changed

File: nodejs/envconfig.json (hunk 17 above)
Comment (girishg4t) [reply nodejs/envconfig.json]: removed

File: nodejs/envconfig.json (hunk 18 above)
Comment (girishg4t) [reply nodejs/envconfig.json]: done

File: python/envconfig.json (hunk 19 above)
Comment (girishg4t) [reply python/envconfig.json]: chnaged

File: python/envconfig.json (hunk 20 above)
Comment (girishg4t) [reply python/envconfig.json]: Changed

File: python/envconfig.json (hunk 21 above)
Comment (girishg4t) [reply python/envconfig.json]: done

**Comment by girishg4t** [comment]
> Hey @girishg4t Left a bunch of comments. @sahil-lakhwani Can you please align with this one and give this another review once Girish has fixed things.
> 
> @girishg4t Can we filter by the status too (Alpha/Beta) etc.?
> 
> Also, I see something missing such as Python has two env versions - 2 & 3 based, similar thing for NodeJs, Go, JVM etc. so we need to make those entries too.

Filter by status can be done, just want to discuss few things
1) Mostly the status will be stable, so will this filter help ?
2) also we are adding different runtime version but we are not showing in ui, where should we use that ?

one's @sahil-lakhwani finished doing changes for all environments we can have a call

File: dotnet/envconfig.json [hunk 30]
```diff
@@ -0,0 +1,26 @@
+[{
+    "version": "1.0.0", 
+    "runtimeVersion" : "1.1.0-runtime", 
+    "kind": "environment", 
+    "status": "Stable", 
+    "name": "Dotnet Environment",
+    "shortDescription": "Fission Dotnet 1.1.0 runtime which uses Kestrel with Nancy to host the internal web server and uses Roslyn to compile the uploaded code.",
```
Comment (sahil-lakhwani) [inline dotnet/envconfig.json:7]:
```suggestion
    "shortDescription": "Fission Dotnet 1.1.0 runtime uses Kestrel with Nancy to host the internal web server and uses Roslyn to compile the uploaded code.",   
```

**Review by sahil-lakhwani — CHANGES_REQUESTED** [review]
hey @girishg4t  On the UI, the check boxes are not being unchecked once selected.
Left few more comments.

File: environments-ui/package-lock.json [hunk 31]
```diff
@@ -0,0 +1,8178 @@
+{
```
[the remark this answers is not in the document]
Comment (girishg4t) [reply environments-ui/package-lock.json:1]: it is kind of go.sum in nodejs, so putting that will not be an issue

File: dotnet/envconfig.json (hunk 30 above)
Comment (girishg4t) [reply dotnet/envconfig.json]: Changed

**Comment by girishg4t** [comment]
> hey @girishg4t On the UI, the check boxes are not being unchecked once selected.
> Left few more comments.

Thanks @sahil-lakhwani , i have corrected the code as per review comment

**Comment by vishal-biyani** [comment]
I see some duplicate between this and https://github.com/fission/environments/pull/8 - how do we plan to take this forward?

**Comment by girishg4t** [comment]
> I see some duplicate between this and #8 - how do we plan to take this forward?
i will merger the code changes by @sahil-lakhwani into my branch and take that forward

**Comment by vishal-biyani** [comment]
Thanks for this @girishg4t - this is a great start 🚀

**Pull request merged**